docs/plugins/private-plugins.md

```diff
@@ -0,0 +1,268 @@
+For more detailed information about settings and configurations, see our [plugin settings documentation](/plugins/define-plugin-settings).
+:::
+
+## Publish Your Plugin
+
+After adding a version to your plugin, configuring your plugin settings, and tagging your repository, it's time to publish your plugin.
+
+1. Go to your Platform Console
+2. Click on "Publish New Version" in your Platform Changes sidebar
+3. Add an optional version message
+4. Click **Publish** to make your changes available
+
+![Publish Your Platform Console Changes](./assets/private-plugins/publish-changes.png)
+
+## Use Private Plugins in Templates and Services
+
+You can use your private plugins in both templates and services, provided the code generator matches your plugin's code generator.
+
+### Include the Plugin in a Template
+
+1. Navigate to the Templates tab in your Platform Console
+2. Click on Create Template
+3. Ensure that your template's code generator matches your private plugin's code generator
+4. In your template's overview page, click on Plugins
+5. Go to the Private Plugins tab on the left sidebar
+6. Click Install on your private plugin
+
+### Include Plugin in a Service
+
+1. Navigate to your service in the Platform Console
+2. Click on the Plugins tab
+3. Look for your private plugin in the Private Plugins section
+   :::note
+   Your plugin will only appear if its code generator matches your service's code generator
+   :::
+4. Click Install to add the plugin to your service
+
+![Include Plugin in a service](./assets/private-plugins/add-private-plugin-to-service.png)
+
+### Configure Plugin Settings
+
+1. Go to the Installed Plugins tab in the left sidebar
+2. Click Settings next to your private plugin
+3. Enter the JSON configuration that matches your plugin's user-facing settings
+
+## Best Practices and Common Pitfalls
+
+### Repository Management
+
+- Keep private plugins in a dedicated git repository separate from services
+- Maintain proper version control with meaningful commits
+
+### Code Generator Compatibility
+
+- Always match plugin and service code generators
+- Verify compatibility before starting development
+
+### Version Control
+
+- Use correct tag format: `PluginId@Version`
+- Always push tags to remote repository that match the tag you set in the UI
+
+### Common Issues and Solutions
+
+- Undefined property errors: Verify plugin versions and publishing
+- Dependency errors: Check `package.json` configuration
+- Build failures: Review logs and ensure proper repository setup
+
+## Next Steps
+
+You successfully created your private plugin.
+It's version-controlled in your Git repository and available for use in both templates and services that match its code generator.
+
+For additional support, contact our support team.
```

Comment:
Let's elaborate on this...
Something like - for more information or in case of issues and questions, use our Discord channel (link) or contact our support team in case you are customers